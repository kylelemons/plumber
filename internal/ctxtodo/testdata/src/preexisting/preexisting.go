// Copyright 2021 Kyle Lemons
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package main

import (
	"context"
	"net/http"
)

func a() {
	_ = context.TODO() // want "Plumb context"
}

func b(ctx context.Context) {
	a()
}

func c() {
	ctx := context.Background()
	a()
	_ = ctx
}

func d(ctx context.Context) {
	_ = context.TODO() // want "Plumb context"
}

func e(r *http.Request) {
	a()

	ctx := context.Background()
	_ = ctx
}

func f(*http.Request) { // want "Name this param if you want plumber to use it"
	a()
}

func g() {
	func(ctx context.Context) {
		a()
	}(nil)
	func(*http.Request) { // want "Name this param if you want plumber to use it"
		a()
	}(nil)
}

func h() {
	a()

	func() {
		var r *http.Request
		a()
		_ = r
	}()

	var r *http.Request
	a()
	_ = r
}

func i(r *http.Request) {
	foo := context.TODO() // want "Plumb context"
	_ = foo
}

func j(r *http.Request) {
	ctx := context.TODO() // want "Plumb context"
	_ = ctx
}

func main() {
	a()
	a()
}

func init() {
	a()
}
