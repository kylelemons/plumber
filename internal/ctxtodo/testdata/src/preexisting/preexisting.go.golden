// Copyright 2021 Kyle Lemons
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package main

import (
	"context"
	"net/http"
)

func a(ctx context.Context) {
	// want "Plumb context"
}

func b(ctx context.Context) {
	a(ctx)
}

func c() {
	ctx := context.Background()
	a(ctx)
	_ = ctx
}

func d(ctx context.Context) {
	// want "Plumb context"
}

func e(r *http.Request) {
	a(r.Context())

	ctx := context.Background()
	_ = ctx
}

func f(*http.Request) { // want "Name this param if you want plumber to use it"
	a(unnamedParam0.Context())
}

func g() {
	func(ctx context.Context) {
		a(ctx)
	}(nil)
	func(*http.Request) { // want "Name this param if you want plumber to use it"
		a(unnamedParam0.Context())
	}(nil)
}

func h(ctx context.Context) {
	a(ctx)

	func() {
		var r *http.Request
		a(r.Context())
		_ = r
	}()

	var r *http.Request
	a(r.Context())
	_ = r
}

func i(r *http.Request) {
	foo := r.Context() // want "Plumb context"
	_ = foo
}

func j(r *http.Request) {
	ctx := r.Context()
	// want "Plumb context"
	_ = ctx
}

func main() {
	ctx := context.Background()
	a(ctx)
	a(ctx)
}

func init() {
	ctx := context.Background()
	a(ctx)
}
